tools/eslint-rules/prefer-proto.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview null objects can be created with `ObjectCreate(null)`, or with
+ *               syntax: `{ __proto__: null }`. This linter rule forces use of
+ *               syntax over ObjectCreate.
+ * @author Jordan Harband <ljharb@gmail.com>
+ */
+'use strict';
+
+// Cribbed from `eslint-module-utils/declaredScope`
+function declaredScope(context, name) {
+  const references = context.getScope().references;
+  const reference = references.find((x) => x.identifier.name === name);
+  if (!reference) return undefined;
+  return reference.resolved.scope.type;
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    messages: {
+      error: 'Use `{ __proto__: null }` instead of `ObjectCreate(null)`.',
+    },
+  },
+  create(context) {
+    return {
+      /* eslint max-len: 0 */
+      'CallExpression[callee.type="Identifier"][callee.name="ObjectCreate"][arguments.0.type="Literal"][arguments.0.value=null][arguments.length=1],\
+      CallExpression[callee.type="MemberExpression"][callee.object.name="Object"][callee.property.name="create"][arguments.0.type="Literal"][arguments.0.value=null][arguments.length=1]\
+      '(node) {
+        if (node.callee.type === 'MemberExpression') {
+          const scope = declaredScope(context, node.callee.object);
+          if (scope !== 'module' && scope !== 'global') {
+            return;
+          }
+        }
```

Comment:
It looks like it misses a lot of occurences (all of them?) https://github.com/nodejs/node/blob/cf6996458b82ec0bdf97209bce380e1483c349fb/benchmark/es/map-bench.js#L28 https://github.com/nodejs/node/blob/c6dabe30832e40dab7fa35fd2ccd0135c74ca091/doc/api/assert.md#L667 https://github.com/nodejs/node/blob/ed116dc3c65a6d568e2de7c09bd1c6f9db2c9b24/test/parallel/test-vm-inherited_properties.js#L24

Reply:
i didn't run the modifications on anything yet, because it takes so long, i was going to let CI find them for me :-)

Reply:
ah, you're right that it's not warning on these files. I'll take another look at it.